src/wallet/rpcwallet.cpp

```diff
@@ -1356,7 +1356,8 @@ static void MaybePushAddress(UniValue & entry, const CTxDestination &dest)
         entry.push_back(Pair("address", addr.ToString()));
 }
 
-void ListTransactions(const CWalletTx& wtx, const string& strAccount, int nMinDepth, bool fLong, UniValue& ret, const isminefilter& filter)
+
+void ListTransactions(const CWalletTx& wtx, const string& strAccount, int nMinDepth, bool fLong, UniValue& ret, const isminefilter& filter,const string& address)
```

Comment:
Given the nature of the issue you are handling I would really like to remove strAccount, which is set to '*' on every  ListTransactions call I see.
Let's maybe crosscheck with @alsala.